crates/adapters/src/transport/adhoc.rs

```diff
@@ -266,21 +266,23 @@ impl InputReader for AdHocInputEndpoint {
                 let mut guard = self.inner.details.lock().unwrap();
                 let details = guard.as_mut().unwrap();
                 let mut num_records = 0;
+                let mut hash = 0;
                 for chunk in chunks {
                     let (mut buffer, errors) = details.parser.parse(&chunk);
                     details.consumer.buffered(buffer.len(), !chunk.len());
                     details.consumer.parse_errors(errors);
                     num_records += buffer.len();
+                    hash ^= buffer.hash();
```

Comment:
I remember you wanted to avoid hashing in each connector. Why did you end up doing it this way?

Reply:
What I really wanted to avoid was implementing hashing differently in each connector; for example, for files, hashing the file data, for Kafka, hashing the messages, and so on. Here, I'm just hashing the buffers using the same code in each time, and it's just a little bit of code sprinkled through each connector. I think that this could be consolidated slightly further but I'm pleased with the uniformity.